segway/gmtk/params.py

```diff
@@ -200,22 +253,52 @@ def get_header_info(self) -> str:
         return " ".join(line)
 
 
-class Mean(OneLineKind):
+class Mean(OneLineArray):
     """
     A single Mean object.
     __init__ and __str__ methods defined in superclass.
     """
     kind = OBJ_KIND_MEAN
 
 
-class Covar(OneLineKind):
+class Covar(OneLineArray):
     """
     A single Covar object.
     __init__ and __str__ methods defined in superclass.
     """
     kind = OBJ_KIND_COVAR
 
 
+class VirtualEvidence:
+    """
+    A Virtual Evidence object, with syntax inferred from the input.master
+    template implementation.
+    Intended to be printed as one line.
+    Attributes:
+        num_segs: int: number of segments for this Virtual Evidence
+        ve_list_filename: str: filename (or C preprocessor macro) containing
+            the Virtual Evidence list
+    """
+    kind = OBJ_KIND_VECPT
+
+    def __init__(self, num_segs: int, ve_list_filename: str):
+        """
+        Initialize a sinlge Virtual Evidence object.
+        :param num_segs: int: number of segments
+        :param ve_list_filename: str: filename containing Virtual Evidence list
+        """
+        self.num_segs = num_segs
+        self.ve_list_filename = ve_list_filename
+
+    def get_header_info(self) -> str:
+        """
+        Return string representation of own information, for header in
+        input.master section.
+        """
+
+        return f"1 {self.num_segs} 2 {self.ve_list_filename} nfs:{self.num_segs} nis:0 fmt:ascii END"
```

Comment:
It would be great if there were more information about what the magic things here mean like "1", "2", "nfs", and "nis"

Reply:
Eric and I unfortunately haven't been able to find information on this object, so we aren't sure either. If you know of any documentation, we'd greatly appreciate it.